src/modules/mentorships/mentorships.controller.ts

```diff
@@ -88,4 +92,56 @@ export class MentorshipsController {
       success: true,
     };
   }
+
+  @Get(':userId/requests')
+  async getMentorshipRequests(
+    @Req() request: Request,
+    @Param('userId') userId: string,
+  ) {
+    const current: User = await this.usersService.findByAuth0Id(
+      request.user.auth0Id,
+    );
+    const user: User = await this.usersService.findById(userId);
+
+    if (!user) {
+      throw new BadRequestException('User not found');
+    }
+
+    // Only an admin or same user can view the requests
+    if (!current._id.equals(user._id) && !current.roles.includes(Role.ADMIN)) {
+      throw new UnauthorizedException(
+        'You are not authorized to perform this operation',
+      );
+    }
+
+    // Get the mentorship requests from and to to that user
+    const mentorshipRequests: Mentorship[] = await this.mentorshipsService.findMentorshipsByUser(
+      userId,
+    );
+
+    // Format the response data
+    let requests = mentorshipRequests.map(item => {
+      const mentorshipSummary = new MentorshipSummaryDto({
+        id: item._id,
+        status: item.status,
+        message: item.message,
+        background: item.background,
+        expectation: item.expectation,
+        date: item.createdAt,
+        isMine: item.mentee.equals(current._id),
+        user: new UserSummaryDto({
+          id: item.mentee._id,
```

Comment:
Mentors and Mentees will use the same endpoint to find out the status of the request, for mentors this response is fine, as we want to show the mentee's data. However, for the mentee we want to show the mentor's info.

We need to check if the current user calling this endpoint is either the mentor or the mentee and send the user's data accordingly.

Reply:
Ah yes, I missed this. I'll update!

Reply:
what about when the admin calls this endpoint? should we populate the mentor or the mentee? also there is potential for this to get confusing when mentor can also be a mentee 🤔 ? should the response instead have `mentor` and `mentee` properties instead of `user`?

Reply:
Very good observation, yes, let's return both users and let the client decide which user to show based on `isMine`.

Reply:
updated!